ingenialink/ethercat/servo.py

```diff
@@ -94,13 +94,27 @@ def __init__(
     ):
         if not pysoem:
             raise pysoem_import_error
-        self.__slave = slave
+        self.__slave: CdefSlave = slave
         self.slave_id = slave_id
         self._connection_timeout = connection_timeout
         self.__emcy_observers: list[Callable[[EmergencyMessage], None]] = []
         self.__slave.add_emergency_callback(self._on_emcy)
         super().__init__(slave_id, dictionary_path, servo_status_listener)
 
+    def check_servo_is_not_in_operational_state(self) -> None:
+        """Checks if the servo is in operational state.
+
+        Raises:
+            ILPDOOperationalError: if servo is in operational state.
+        """
+        if self.slave is None or not pysoem:
+            return
+        self.slave.read_state()
+        if self.slave.state in [pysoem.OP_STATE, pysoem.SAFEOP_STATE]:
```

Comment:
Maybe we should check this the other way around (i.e. only allow to modify the PDOs in PreOp) because if the slave is in the SAFEOP_ERROR this check will fail.